pkg/sac/scope_key.go

```diff
@@ -85,10 +85,16 @@ func (k ResourceScopeKey) String() string {
 }
 
 // ResourceScopeKeys wraps the given resources in a scope key slice.
+// Note: The returned scope keys _may_ be greater than the given resources,
+// since replacing resources will be taken into account and conditionally added
+// to the returned scope keys.
```

Comment:
I'd also explicitly say that you think that the broadening is fine because of the contexts this function is used in.